components/TodoList.js

```diff
@@ -1,37 +1,41 @@
-import  { useState } from 'react';
+import { useState } from 'react';
 
 function TodoList() {
-  const [todos, setTodos] = useState([]);  
-  const [newTodo, setNewTodo] = useState('');  
+  const [todos, setTodos] = useState([]);
+  const [newTodo, setNewTodo] = useState('');
 
-  function handleAddTodo() {
+  function addNewTodo() {
     if (newTodo.trim() !== '') {
-      
       setTodos([...todos, newTodo]);
       setNewTodo('');
     }
   }
 
+  function removeTodo(index) {
+    const newTodos = todos.filter((_, i) => i !== index);
+    setTodos(newTodos);
+  }
+
   return (
     <div>
       <h2>Tasks</h2>
-      
-      <input
-        type="text"
-        value={newTodo}
-        onChange={(e) => setNewTodo(e.target.value)}
-      />
-      
-      <button onClick={handleAddTodo}>add new Task</button>
-      
+      <input type="text" value={newTodo} onChange={(e) => setNewTodo(e.target.value)} />
+      <button onClick={addNewTodo}>Add</button>
       <ul>
         {todos.map((todo, index) => (
-          <li key={index}>{todo}</li>
+          <li key={index}>
+            {todo}
+            <button onClick={() => removeTodo(index)}>Delete</button>
+          </li>
```

Comment:
The `index` is not enough for the `key` prop. In fact, using the `index` is exactly the same as not assigning a value to the `key` at all. You can [read more about it here](https://react.dev/learn/rendering-lists#rules-of-keys).

Use a stable, unique id instead. In your case, either use the `todo` itself or have your array contain objects with an id and a text. In that case, you'd need to generate an id for every new entry, but it's the recommended approach.

**Additionally,** you will need to pass the id to the `removeTodo` function, instead of passing the `index`. An id is a better discriminator and it's stable across render cycles.